@xmatthias  I added most of your requested changes.

What I couldn't add was using `get_analyzed_dataframe()`.
I got this error:
```shell
Traceback (most recent call last):
  File "/home/joe/projects/freqtrade/freqtrade/commands/trade_commands.py", line 19, in start_trading
    worker.run()
  File "/home/joe/projects/freqtrade/freqtrade/worker.py", line 74, in run
    state = self._worker(old_state=state)
  File "/home/joe/projects/freqtrade/freqtrade/worker.py", line 111, in _worker
    self._throttle(func=self._process_running, throttle_secs=self._throttle_secs)
  File "/home/joe/projects/freqtrade/freqtrade/worker.py", line 132, in _throttle
    result = func(*args, **kwargs)
  File "/home/joe/projects/freqtrade/freqtrade/worker.py", line 145, in _process_running
    self.freqtrade.process()
  File "/home/joe/projects/freqtrade/freqtrade/freqtradebot.py", line 184, in process
    self.exit_positions(trades)
  File "/home/joe/projects/freqtrade/freqtrade/freqtradebot.py", line 693, in exit_positions
    if trade.open_order_id is None and trade.is_open and self.handle_trade(trade):
  File "/home/joe/projects/freqtrade/freqtrade/freqtradebot.py", line 799, in handle_trade
    if self._check_and_execute_sell(trade, sell_rate, buy, sell):
  File "/home/joe/projects/freqtrade/freqtrade/freqtradebot.py", line 934, in _check_and_execute_sell
    should_sell = self.strategy.should_sell(
  File "/home/joe/projects/freqtrade/freqtrade/strategy/interface.py", line 518, in should_sell
    stoplossflag = self.stop_loss_reached(current_rate=current_rate, trade=trade,
  File "/home/joe/projects/freqtrade/freqtrade/strategy/interface.py", line 579, in stop_loss_reached
    stop_loss_value = strategy_safe_wrapper(self.custom_stoploss, default_retval=None
  File "/home/joe/projects/freqtrade/freqtrade/strategy/strategy_wrapper.py", line 33, in wrapper
    raise StrategyError(str(error)) from error
freqtrade.exceptions.StrategyError: datetime.datetime(2021, 3, 4, 14, 6, 28, 715919, tzinfo=datetime.timezone.utc)
```
when running it with `freqtrade trade --dry-run -s MyAwesomeStrategy`. I updated my other PR acordinlgy, so you could test it on your mchine if needed: https://github.com/freqtrade/freqtrade-strategies/pull/130 .

--- a/docs/strategy-advanced.md
+++ b/docs/strategy-advanced.md
@@ -234,8 +234,7 @@ class AwesomeStrategy(IStrategy):
 
 Imagine you want to use `custom_stoploss()` to use a trailing indicator like e.g. "ATR"
 
-See: (Storing custom information using DatetimeIndex from `dataframe`
-)[WIP] on how to store the indicator into `custom_info`
+See: "Storing custom information using DatetimeIndex from `dataframe`" example above) on how to store the indicator into `custom_info`
 
 ``` python
 from freqtrade.persistence import Trade

--- a/docs/strategy-advanced.md
+++ b/docs/strategy-advanced.md
@@ -188,6 +188,7 @@ See: (Storing custom information using DatetimeIndex from `dataframe`
 
 ``` python
 from freqtrade.persistence import Trade
+from freqtrade.state import RunMode
 
 class AwesomeStrategy(IStrategy):
 
@@ -200,12 +201,23 @@ class AwesomeStrategy(IStrategy):
 
         result = 1
         if self.custom_info[pair] is not None and trade is not None:
-            atr = self.custom_info[pair].loc[current_time]['atr']
-            if (atr is not None):
+            # using current_time directly (like below) will only work in backtesting.
+            # so check "runmode" to make sure that it's only used in backtesting
+            if(self.dp.runmode == RunMode.BACKTEST):
+              relative_sl = self.custom_info[pair].loc[current_time]['atr]
+            # in live / dry-run, it'll be really the current time
+            else:
+              # but we can just use the last entry to get the current value
+              relative_sl =  self.custom_info[pair]['atr].iloc[ -1 ]
+
+            if (relative_sl is not None):
+                print("Custom SL: {}".format(relative_sl))
                 # new stoploss relative to current_rate
-                new_stoploss = (current_rate-atr)/current_rate
+                new_stoploss = (current_rate-relative_sl)/current_rate
                 # turn into relative negative offset required by `custom_stoploss` return implementation
                 result = new_stoploss - 1
+
+        print("Result: {}".format(result))
         return result
 ```
 

--- a/docs/strategy-advanced.md
+++ b/docs/strategy-advanced.md
@@ -46,15 +46,19 @@ class AwesomeStrategy(IStrategy):
 
 Imagine you need to store an indicator like `ATR` or `RSI` into `custom_info`. To use this in a meaningful way, you will not only need the raw data of the indicator, but probably also need to keep the right timestamps.
 
+```python
+import talib.abstract as ta
 class AwesomeStrategy(IStrategy):
     # Create custom dictionary
     custom_info = {}
 
     def populate_indicators(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
-        # add indicator mapped to correct DatetimeIndex to custom_info
         # using "ATR" here as example
+        dataframe['atr'] = ta.ATR(dataframe)
+        # add indicator mapped to correct DatetimeIndex to custom_info
         self.custom_info[metadata['pair']] = dataframe[['date', 'atr']].copy().set_index('date')
         return dataframe
+```
 
 See `custom_stoploss` examples below on how to access the saved dataframe columns
 

